modules/Oculus_module/src/RobotControlHelper.cpp

```diff
@@ -0,0 +1,279 @@
+/**
+ * @file RobotControlHelper.cpp
+ * @authors Giulio Romualdi <giulio.romualdi@iit.it>
+ * @copyright 2018 iCub Facility - Istituto Italiano di Tecnologia
+ *            Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ * @date 2018
+ */
+
+// iDynTree
+#include <iDynTree/Core/Utils.h>
+
+#include <RobotControlHelper.hpp>
+#include <Utils.hpp>
+
+bool RobotControlHelper::configure(const yarp::os::Searchable& config, const std::string& name,
+                                   bool isMandatory)
+{
+    m_isMandatory = isMandatory;
+
+    // robot name: used to connect to the robot
+    std::string robot;
+    robot = config.check("robot", yarp::os::Value("icubSim")).asString();
+
+    // get all controlled icub parts from the resource finder
+    std::vector<std::string> iCubParts;
+    yarp::os::Value* iCubPartsYarp;
+    if (!config.check("remote_control_boards", iCubPartsYarp))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to find remote_control_boards into "
+                    "config file.";
+        return false;
+    }
+    if (!YarpHelper::yarpListToStringVector(iCubPartsYarp, iCubParts))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to convert yarp list into a vector of "
+                    "strings.";
+        return false;
+    }
+
+    // open the remotecontrolboardremepper YARP device
+    yarp::os::Property options;
+    yarp::os::Value* axesListYarp;
+    if (!config.check("joints_list", axesListYarp))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to find joints_list into config file.";
+        return false;
+    }
+
+    if (!YarpHelper::yarpListToStringVector(axesListYarp, m_axesList))
+    {
+        yError() << "[RobotControlHelper::configure] Unable to convert yarp list into a "
+                    "vector of strings.";
+        return false;
+    }
+
+    options.put("device", "remotecontrolboardremapper");
+    YarpHelper::addVectorOfStringToProperty(options, "axesNames", m_axesList);
+
+    // prepare the remotecontrolboards
+    yarp::os::Bottle remoteControlBoards;
+    remoteControlBoards.clear();
+    yarp::os::Bottle& remoteControlBoardsList = remoteControlBoards.addList();
+    for (auto iCubPart : iCubParts)
+        remoteControlBoardsList.addString("/" + robot + "/" + iCubPart);
+
+    options.put("remoteControlBoards", remoteControlBoards.get(0));
+    options.put("localPortPrefix", "/" + name + "/remoteControlBoard");
+    yarp::os::Property& remoteControlBoardsOpts = options.addGroup("REMOTE_CONTROLBOARD_OPTIONS");
+    remoteControlBoardsOpts.put("writeStrict", "on");
+
+    m_actuatedDOFs = m_axesList.size();
+
+    // open the device
+    if (!m_robotDevice.open(options) && m_isMandatory)
```

Comment:
Notice that in this way it returns false every time ``m_isMandatory`` is false. Anyway if the ``open`` returns ``false`` is always worthwhile to throw an error.

Reply:
Is this the call failing when some finger is in ``NOT CONFIGURED`` status?